episodes/simpleitk.md

```diff
@@ -0,0 +1,1198 @@
+    title=fixed("Confidence connected"))
+```
+
+Since we have available also another MRI scan of the same patient, T2-weighted, we can try to further improve the segmentation.
+We first load a T2 image from the same person and combine it with the T1 image to create a vector image. This region growing algorithm is similar to the previous one, `ConfidenceConnected`, and allows the user to implicitly specify the threshold bounds based on the statistics estimated from the seed points. The main difference is that in this case we are using the Mahalanobis and not the intensity difference.
+
+```python
+img_T2 = sitk.ReadImage("episodes/data/A1_grayT2.nrrd")
+img_multi = sitk.Compose(img_T1, img_T2)
+seg = sitk.VectorConfidenceConnected(img_multi, seedList=[seed],
+                                             numberOfIterations=1,
+                                             multiplier=2.5,
+                                             initialNeighborhoodRadius=1)
+seg_img = sitk.LabelOverlay(img_T1_255, seg)
+
+interact(
+    display_images,
+    image_z=(0,img_T1.GetSize()[2]-1),
+    npa = fixed(sitk.GetArrayViewFromImage(seg_img)),
+    title=fixed("Vector confidence connected"))
+```
+
+![](episodes/fig/reg_grow.png){alt='Region growing segmentations.'}
+
+#### Clean up
+
+Use of low level segmentation algorithms such as region growing is often followed by a clean up step. In this step we fill holes and remove small connected components. Both of these operations are achieved by using binary morphological operations, opening (`BinaryMorphologicalOpening`) to remove small connected components and closing (`BinaryMorphologicalClosing`) to fill holes.
+
+SITK supports several shapes for the structuring elements (kernels) including:
+
+- sitkAnnulus
+- sitkBall
+- sitkBox
+- sitkCross
+
+The size of the kernel can be specified as a scalar (same for all dimensions) or as a vector of values, size per dimension.
+
+The following code cell illustrates the results of such a clean up, using closing to remove holes in the original segmentation.
+
+```python
+seg = sitk.ConfidenceConnected(img_T1, seedList=[seed],
+                                   numberOfIterations=0,
+                                   multiplier=2,
+                                   initialNeighborhoodRadius=1,
+                                   replaceValue=1)
+
+vectorRadius=(1,1,1)
+kernel=sitk.sitkBall
+seg_clean = sitk.BinaryMorphologicalClosing(seg, vectorRadius, kernel)
+seg_img_clean = sitk.LabelOverlay(img_T1_255, seg_clean)
+
+interact(
+    display_images,
+    image_z=(0,img_T1.GetSize()[2]-1),
+    npa = fixed(sitk.GetArrayViewFromImage(seg_img_clean)),
+    title=fixed("Confidence connected after morphological closing"))
+```
+
+![](episodes/fig/reg_grow_cleaned.png){alt='Confidence connected after morphological closing.'}
+
+### Level-Set Segmentation
+
+There are a variety of level-set based segmentation filter available in ITK:
```

Comment:
What is missing here is a sentence to tell me what level set segmentation means. I still have only a vague idea that it has something to do with topology. If it's too complicated to explain, maybe a link to a paper or textbook.